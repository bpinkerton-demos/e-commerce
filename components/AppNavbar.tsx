import {useSession} from "next-auth/react";
import {Container, Dropdown, Image, Nav, Navbar, NavDropdown} from "react-bootstrap";
import {BoxArrowInRight, Cart, Person} from "react-bootstrap-icons";




export default function AppNavbar(){
    const {data: session} = useSession()
    return (
        <Navbar fixed={"top"} bg={"dark"} variant={"dark"} expand={"sm"}>
            <Container fluid>
                <Navbar.Brand href={"/"}>E-Commerce</Navbar.Brand>
                <Navbar.Toggle className={"float-right"}/>
                <Navbar.Collapse id={"responsive-navbar-nav"}>
                    <Nav className={"ms-auto"}>
                        {session && (<AuthNav user={session.user as AuthUser}/>)}
                        {!session && (<NoAuthNav/>)}
                    </Nav>
                </Navbar.Collapse>
            </Container>
        </Navbar>
    )
}

function NoAuthNav(){
    return (
        <Nav.Link className={"float-right"} href={`/api/auth/signin`}>Login <Person size={20}/></Nav.Link>
    )
}


type AuthUser = {
    email: string,
    image: string,
    name: string
}

interface AuthNavProps{
    user: AuthUser
}


function AuthNav({user}: AuthNavProps){

    return (
        <>

            <UserCart />
            <NavDropdown id={"user-dropdown"} menuVariant={"dark"} title={
                <Image src={user.image} alt={"image of logged in user"} roundedCircle={true} height={40}/>
            }  flip >
                <Dropdown.ItemText >{user.name}</Dropdown.ItemText>
                <Dropdown.ItemText >{user.email}</Dropdown.ItemText>
                <Dropdown.Item className={"float-right"} href={`/api/auth/signout`}>Logout <BoxArrowInRight size={20}/></Dropdown.Item>
            </NavDropdown>
        </>
    )
}

function UserCart(){
    return (
        <>
            <NavDropdown className={"cart-dropdown"} menuVariant={"dark"} title={
                <Nav.Link><Cart size={20}></Cart></Nav.Link>
            }>
                <Dropdown.Item>Checkout</Dropdown.Item>
            </NavDropdown>
        </>

    )
}