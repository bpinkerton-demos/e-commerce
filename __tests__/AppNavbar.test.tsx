import {render} from '@testing-library/react'
import AppNavbar from "../components/AppNavbar";
import {Session} from "next-auth";

jest.mock("next-auth/react", () => {
    const originalModule = jest.requireActual('next-auth/react');
    const mockSession: Session = {
        expires: "1",
        user: { email: "test@test.com", name: "Test", image: "test.png" },
    };
    return {
        _esModule: true,
        ...originalModule,
        useSession: jest.fn(() =>{
            return {data: mockSession, status: 'authenticated'}
        })
    }
})


it('Renders navbar', () => {
    const {container} = render(<AppNavbar/>)
    expect(container).toMatchSnapshot()
})