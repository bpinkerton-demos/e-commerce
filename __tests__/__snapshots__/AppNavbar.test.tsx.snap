// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders navbar 1`] = `
<div>
  <nav
    class="navbar navbar-expand-sm navbar-dark bg-dark fixed-top"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        E-Commerce
      </a>
      <button
        aria-label="Toggle navigation"
        class="float-right navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="ms-auto navbar-nav"
        >
          <div
            class="cart-dropdown nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="dropdown-toggle nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              <a
                class="nav-link"
                href="#"
                role="button"
                tabindex="0"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  viewBox="0 0 16 16"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 1.5A.5.5 0 0 1 .5 1H2a.5.5 0 0 1 .485.379L2.89 3H14.5a.5.5 0 0 1 .491.592l-1.5 8A.5.5 0 0 1 13 12H4a.5.5 0 0 1-.491-.408L2.01 3.607 1.61 2H.5a.5.5 0 0 1-.5-.5zM3.102 4l1.313 7h8.17l1.313-7H3.102zM5 12a2 2 0 1 0 0 4 2 2 0 0 0 0-4zm7 0a2 2 0 1 0 0 4 2 2 0 0 0 0-4zm-7 1a1 1 0 1 1 0 2 1 1 0 0 1 0-2zm7 0a1 1 0 1 1 0 2 1 1 0 0 1 0-2z"
                  />
                </svg>
              </a>
            </a>
          </div>
          <div
            class="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="dropdown-toggle nav-link"
              href="#"
              id="user-dropdown"
              role="button"
              tabindex="0"
            >
              <img
                alt="image of logged in user"
                class="rounded-circle"
                height="40"
                src="test.png"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
